commit 4c339b2c51094d3be01338eef9ba6c088e65ce5c
Author: Trent Mick <trentm@gmail.com>
Date:   2019-05-16T15:59:18-07:00 (5 months ago)
    
    TRITON-1677 broke callback handling in vmapi lib/endpoints/vms.js can lead to double-callback

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index 832b259..4acffca 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -1179,7 +1179,9 @@ function _cnapiServerNotFoundError(err) {
  */
 function createRoleTags(req, cb) {
     var error, message;
+    var i;
     var roleTags = req.params.role_tags;
+    var roleTag;
 
     if (!Array.isArray(roleTags) || Object.keys(roleTags).length === 0) {
         message = 'Must be an array of UUIDs';
@@ -1188,15 +1190,16 @@ function createRoleTags(req, cb) {
         return;
     }
 
-    roleTags.forEach(function (roleTag) {
+    for (i = 0; i < roleTags.length; ++i) {
+        var roleTag = roleTags[i];
         if (!common.validUUID(roleTag)) {
             message = roleTag + ' is not a UUID';
             error = [ errors.invalidUuidErrorsElem('role_tags', message) ];
-            // XXX This is broken. It can `cb` multiple times.
             cb(new errors.ValidationFailedError('Invalid VM parameters',
                 error));
+            return;
         }
-    });
+    }
 
     req.app.moray.putVmRoleTags(req.params.uuid, roleTags, cb);
 }
